lms/djangoapps/certificates/api.py

```diff
@@ -576,25 +586,29 @@ def get_certificate_footer_context():
 def certificates_viewable_for_course(course):
     """
     Returns True if certificates are viewable for any student enrolled in the course, False otherwise.
+
+    Arguments:
+        course (CourseOverview or course descriptor): The course to check if certificates are viewable
+
+    Returns:
+        boolean: whether the certificates are viewable or not
     """
-    if course.self_paced:
-        return True
-    if (
-        course.certificates_display_behavior in ('early_with_info', 'early_no_info')
-        or course.certificates_show_before_end
-    ):
-        return True
-    if (
-        course.certificate_available_date
-        and course.certificate_available_date <= datetime.now(UTC)
-    ):
-        return True
-    if (
-        course.certificate_available_date is None
-        and course.has_ended()
-    ):
-        return True
-    return False
+
+    # The CourseOverview contains validation logic on the certificates_display_behavior and certificate_available_date
+    # fields. Thus, we prefer to use the CourseOverview, but will fall back to the course in case the CourseOverview is
+    # not available.
+    if not isinstance(course, CourseOverview):
```

Comment:
Should this live in the certificates Django app? Doesn't this violate OEP-49?

Reply:
Yeah, this isn't my favourite thing. My original version exposed a function where you just pass in the field values instead of the object. (https://github.com/edx/edx-platform/pull/28286/files#diff-0f3ac000f80eebb6a8b7cbdae7496934f9e06a9d75fee952725d0f3d324bfbb9L757) that way there is no special knowledge needed. But this code already existed and was used by far more things and I didn't want to another chunk of the codebase to this PR. We can follow with it if we want, but I'm going to leave a bad taste in my mouth for sanity for now.

Reply:
Thanks, I figured I'd at least ask. I trust your instincts and don't want to create more work for you at this point or cause this story to linger any longer. I'm sure we can always return later and address it when needed.